app/src/main/java/com/example/dispatcher/presentation/profile/viewModel/ProfileViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.dispatcher.presentation.profile.viewModel
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.dispatcher.presentation.homepage.model.Article
+import com.example.dispatcher.domain.homepage.repo.ArticleRepository
+
+class ProfileViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val articleRepository = ArticleRepository(application)
+    private lateinit var articlesList: MutableList<Article>
+
+    private val authorsLiveData = MutableLiveData<String>()
+
+    init {
+        articlesList = articleRepository.getArticles().toMutableList()
+        updateAuthors()
+    }
+
+    fun getAuthorsLiveData(): LiveData<String> {
+        return authorsLiveData
+    }
+
+    fun addAuthor(author: String) {
+        val currentTitles = authorsLiveData.value ?: ""
```

Comment:
currentTitles ?? 
don't just copy paste code you think is similar, this causes problems as in this one.